pkg/config/config.go

```diff
@@ -272,6 +287,9 @@ func initConfig(config Config) {
 	config.BindEnvAndSetDefault("dogstatsd_so_rcvbuf", 0)
 	config.BindEnvAndSetDefault("dogstatsd_metrics_stats_enable", false)
 	config.BindEnvAndSetDefault("dogstatsd_tags", []string{})
+	config.BindEnvAndSetDefault("dogstatsd_mapper_cache_size", 1000)
+	config.SetKnown("dogstatsd_mapper_profiles")
```

Comment:
Seems not possible, but I'm not that familiar with viper:

```
=== RUN   TestMappingCases/Simple_OK_case
'dogstatsd_mapper_profiles: []'
Error: Could not parse dogstatsd_mapper_profiles: 1 error(s) decoding:
* '[0]' expected a map, got 'string'
```

It seems container is handling the `datadog.yaml` that way: https://github.com/DataDog/k8s-resources/blob/b1361cb1d7141660a9ff705c2a2243543adfe8f8/k8s/datadog-agent/templates/configmap.yaml#L45-L59

Reply:
Yes internally we are mounting the configuration file in which case it would work without issues. That said mounting the file is not a requirement and probably not done in all setups. 

@mfpierre How do you deal with complex configuration options mapping to custom objects in env vars ? (Might be worth asking them directly as it will probably get lost in the notification spam)